scripts/smg-html.py

```diff
@@ -0,0 +1,208 @@
+        dotCmd = "dot -Tpng " + filePath + " -o " + destPath
+        os.system(dotCmd)
+        # create new dot list item
+        parseName(name)
+
+# open JSON file and correct it to be valid
+os.chdir(dirOutput)
+with open("visualisationSchema.json", "rb+") as mapFile:
+    mapFile.seek(-2, os.SEEK_END)
+    if mapFile.readline() == ",\n":
+        mapFile.seek(-2, os.SEEK_END)
+        mapFile.truncate()
+        with open("visualisationSchema.json", "a") as mapFile:
+            mapFile.write("]")
+
+
+# class for tree representing analysis map
+class SMGState(object):
+    def __init__(self):
+        self.number = None
+        self.fileName = None
+        self.succ = None
+        self.pred = None
+
+
+# parse json file
+with open("visualisationSchema.json") as mapFile:
+    states = json.load(mapFile)
+
+
+def getDotName(number):
+    for dot in dotList:
+        if dot.number == number:
+            return dot.name
+    return None
+
+# create SMG states into dictionary from parsed JSON
+smgDict = {}
+for state in states:
+    if state['state'] not in smgDict:
+        # create new SMG state object
+        smg = SMGState()
+        smg.number = state['state']
+        smg.succ = state['succ']
+        smg.succ[:] = [succ for succ in smg.succ if not succ == smg.number]  # remove self-loops
+        smg.fileName = getDotName(smg.number)
+        # put SMG object into dictionary
+        smgDict[smg.number] = smg
+    else:
+        # object for given state already exists - add new successors
+        smgDict[state['state']].succ.extend(state['succ'])
+
+# create predecessors
+for number, smg in smgDict.items():
+    if smg.succ:
+        for succ in smg.succ:
+            if succ in smgDict:
+                smgDict[succ].pred = number
+
+# delete states which do not have dot files created
+for number, smg in sorted(smgDict.items()):
+    if not smg.fileName:
+        # edit predecessor succesors
+        if smg.pred in smgDict:
+            smgDict[smg.pred].succ.remove(number)
+            smgDict[smg.pred].succ.extend(smg.succ)
+        # edit successors predecesors
+        for succ in smg.succ:
+            if succ in smgDict:
+                smgDict[succ].pred = smg.pred
+        # delete state
+        del smgDict[number]
+
+# create ending states (they are not in the dictionary, but have corresponding dot)
+for number, smg in smgDict.items():
+    for succ in smg.succ:
+        if succ not in smgDict:
+            # create new SMG state item
+            newSmg = SMGState()
+            newSmg.number = succ
+            newSmg.fileName = getDotName(newSmg.number)
+            newSmg.pred = number
+            # add new SMG to dictionary
+            smgDict[newSmg.number] = newSmg
+
+# print the final tree structure
+# for number, smg in sorted(smgDict.items()):
+#     print smg.number + ":"
+#     print smg.fileName
+#     if smg.succ:
+#         for succ in smg.succ:
+#             print succ
+#     print "\n"
```

Comment:
Please do not commit commented-out code. Such code can either be part of some debugging output (triggered by some debug option) or not checked in at all.